pkg/query-service/utils/format.go

```diff
@@ -389,3 +389,10 @@ func NormalizeMap(data map[string]uint64) map[string]float64 {
 
 	return normalized
 }
+
+func FormatFloat(f float64) string {
+	if f == math.Trunc(f) {
+		return fmt.Sprintf("%.0f", f)
+	}
+	return fmt.Sprintf("%g", f)
```

Comment:
As I understand, this change is trying to format the non-decimal floats. The decimal floats will still be formatted as they are now. Is that the desired outcome https://go.dev/play/p/trAh_ppYSRZ?

There will be people complaining the same about floats with decimal floating points?

Reply:
https://pkg.go.dev/fmt#:~:text=The%20default%20format,p%0Apointer%3A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%25p

Reply:
Yeah, if it's an actual floating point then it's not doing anything else, but if I use %f and the number of zeros need to be specified and if there are less zeros in the actual number then extras will be added.

Reply:
We can use https://pkg.go.dev/strconv#FormatFloat?

Reply:
what value of `fmt` are you suggesting to pass in the above ?

Reply:
(value, `f`, -1, 64).

Reply:
interesting, thanks.

Reply:
Made the change